core/src/main/java/org/infinispan/interceptors/MarshalledValueInterceptor.java

```diff
@@ -258,7 +258,9 @@ public Object visitReplaceCommand(InvocationContext ctx, ReplaceCommand command)
       return new AbstractDelegatingEntryCacheSet<K, V>(getCacheWithFlags(cache, command), set) {
          @Override
          public CloseableIterator<CacheEntry<K, V>> iterator() {
-            return new CloseableIteratorMapper<>(super.iterator(), e -> unwrapEntry(e, ctx));
+            // We pass a null ctx, since this is always local invocation - if it was remote it would use
+            // DistributionBulkInterceptor
```

Comment:
The wording may be confusing here I guess.  But the problem is that it can also be invoked remotely - determined by the terminal operation.  However we always want this unwrapped even for the remote side.

I can try to fix this a bit.